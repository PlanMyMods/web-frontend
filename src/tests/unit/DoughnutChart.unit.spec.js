import { shallowMount } from "@vue/test-utils";
import DoughnutChart from "@/components/DoughnutChart.vue";

describe("DoughnutChart", () => {
  const wrapper = shallowMount(DoughnutChart, {
    propsData: {
      assessmentName: ["Assessment 1", "Assessment 2"],
      weightage: [10, 20],
    },
  });
  test("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("should render the correct labels", () => {
    expect(wrapper.vm.assessmentName).toEqual(["Assessment 1", "Assessment 2"]);
  });
  test("should render the correct weightage", () => {
    expect(wrapper.vm.weightage).toEqual([10, 20]);
  });
  test("should render a canvas element", () => {
    expect(wrapper.find("canvas").exists()).toBe(true);
  });
});
