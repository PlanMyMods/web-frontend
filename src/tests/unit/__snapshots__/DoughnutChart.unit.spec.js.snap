// Vitest Snapshot v1

exports[`DoughnutChart > should match snapshot 1`] = `"<canvas id=\\"myChart\\" width=\\"300\\" height=\\"300\\"></canvas>"`;
